number_search.py

```diff
@@ -2,32 +2,33 @@
 #  in article.txt
 # use regex to match numbers and fractions
 import re
-frac_pattern = r'(-?\d+\\[A-Za-z]+{\d+}{\d+} |-?\\[A-Za-z]+{\d+}{\d+})'
-number_pattern = r"([^a-zA-Z{./\\}]\d+ |\d+\.\d+|[:]\d+|\d+[:])"
+frac_pattern = r'(-?\d+\\tinyfrac{\d+}{\d+}[^A-Za-z]|-?\\frac{\d+}{\d+}[^A-Za-z])'
+number_pattern = r'([^a-zA-Z{./\\}]\d+ |\d+\.\d+| \d+[^/:\\)-}])'
+numbers = []
+fractions = []
 
 
 def count_numbers(text):
     """
         counts whole number and decimals i.e -ve and +ve whole numbers decimals
         TODO update doc strings
     """
-    count = -1
+
     with open(text, "r") as f:
         article = f.read()
```

Comment:
try using split() to convert the string to list for better capture because RE have no stop or continue... but if you split by space u get list of string and can use each string to match pattern and can use $ to make sure the number not followed by invalid character check read me again